packages/suite/src/components/suite/modals/ReduxModal/CopyContractAddress.tsx

```diff
@@ -0,0 +1,49 @@
+import styled from 'styled-components';
+import { copyToClipboard } from '@trezor/dom-utils';
+import { notificationsActions } from '@suite-common/toast-notifications';
+
+import { Translation } from 'src/components/suite';
+import { useDispatch } from 'src/hooks/suite/useDispatch';
+import { Button } from '@trezor/components';
+import { DialogModal } from '../Modal/DialogRenderer';
+
+const StyledModal = styled(DialogModal)`
+    width: 600px;
+`;
+
+interface CopyContractAddressModalProps {
+    contract: string;
+    onCancel: () => void;
+}
+
+export const CopyContractAddressModal = ({ contract, onCancel }: CopyContractAddressModalProps) => {
+    const dispatch = useDispatch();
+
+    const onCopyContractAddress = () => {
+        const result = copyToClipboard(contract);
+        if (typeof result !== 'string') {
+            dispatch(notificationsActions.addToast({ type: 'copy-to-clipboard' }));
+        }
+        onCancel();
+    };
+
+    return (
+        <StyledModal
+            isCancelable
+            onCancel={onCancel}
+            icon="warningTriangleLight"
```

Comment:
there is cursor `pointer` on icon and it is weird @jvaclavik

Reply:
@AdamSchinzel I pushed to your PR fix for the cursor bug: https://github.com/trezor/trezor-suite/pull/12934/commits/86194c9d6c551a190e90308687730278c81b59a1